### PR TITLE
8339316: Test runtime/exceptionMsgs/NoClassDefFoundError/NoClassDefFoundErrorTest.java fails after  JDK-8338257

### PR DESCRIPTION
**Pull request by dholmes-ora**
In JDK-8338257 I overlooked updating the callers of `UTF8::is_legal_utf8` to pass a `size_t` length parameter. In some cases the length was explicitly cast to `int` and in the test case in question (with `-Xcheck:jni`) this caused integer overflow to a negative value which then became an exceedingly large `size_t` value and we then tried to do utf8 validation on random bytes.

Testing:
- failing test
- tiers 1-4

Thanks

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339316](https://bugs.openjdk.org/browse/JDK-8339316): Test runtime/exceptionMsgs/NoClassDefFoundError/NoClassDefFoundErrorTest.java fails after  JDK-8338257 (**Bug** - P3)

### Reviewers
 * [Johan Sjölen](https://openjdk.org/census#jsjolen) (@jdksjolen - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20804/head:pull/20804` \
`$ git checkout pull/20804`

Update a local copy of the PR: \
`$ git checkout pull/20804` \
`$ git pull https://git.openjdk.org/jdk.git pull/20804/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20804`

View PR using the GUI difftool: \
`$ git pr show -t 20804`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20804.diff">https://git.openjdk.org/jdk/pull/20804.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20804#issuecomment-2323577760)

### CODE REVIEW

File: src/hotspot/share/classfile/systemDictionary.cpp [hunk 1]
```diff
@@ -285,8 +285,7 @@ Symbol* SystemDictionary::class_name_symbol(const char* name, Symbol* exception,
     return nullptr;
   }
   // Callers should ensure that the name is never an illegal UTF8 string.
-  assert(UTF8::is_legal_utf8((const unsigned char*)name,
-                             static_cast<int>(name_len), false),
+  assert(UTF8::is_legal_utf8((const unsigned char*)name, name_len, false),
```
Comment (coleenp) [inline src/hotspot/share/classfile/systemDictionary.cpp:288]: Is there where the > INT_MAX length got in from JNI?

Comment (dholmes-ora) [reply src/hotspot/share/classfile/systemDictionary.cpp:288]: No, the failing path was in jniCheck.cpp. In this code we have already rejected strings > `Symbol::max_length`.

**Review by jdksjolen — APPROVED** [review]
LGTM, thank you.

**Comment by dholmes-ora** [comment]
Thanks for the review @jdksjolen !

**Review by coleenp — APPROVED** [review]
Looks good.

**Comment by dholmes-ora** [comment]
Thanks @coleenp !

/integrate

**Pull request closed without merging**